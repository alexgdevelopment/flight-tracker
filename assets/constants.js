export const SEARCH_TYPE = Object.freeze({
  AIRPORT: 'search_type_airport',
  FLIGHT: 'search_type_flight',
  NOT_SELECTED: 'search_type_not_selected',
})

export const AIRPORT_DIRECTION = Object.freeze({
  ARRIVING: 'airport_direction_arriving',
  DEPARTING: 'airport_direction_departing',
  NOT_SELECTED: 'airport_direction_not_selected',
})
