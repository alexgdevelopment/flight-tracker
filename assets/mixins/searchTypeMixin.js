import { SEARCH_TYPE } from '../constants'
const searchTypeMixin = {
  data() {
    return {
      searchTypeVar: SEARCH_TYPE,
    }
  },
}

export default searchTypeMixin
