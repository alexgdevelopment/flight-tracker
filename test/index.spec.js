import { Store } from 'vuex'
import { renderWithVuetify } from './helpers'
import index from '@/pages/index.vue'
import searchTypeMixin from '@/assets/mixins/searchTypeMixin.js'
import { SEARCH_TYPE } from '@/assets/constants.js'

describe('The "index" page', () => {
  let wrapper = null
  let store
  let state

  describe('when there is no search type selected', () => {
    beforeEach(() => {
      state = {
        searchType: SEARCH_TYPE.NOT_SELECTED,
      }
      store = new Store({
        state,
      })
      wrapper = renderWithVuetify(index, {
        store,
        mixins: [searchTypeMixin],
        stubs: {
          SearchTypeSelector: {
            template: `<div data-testid='search-type-selector' />`,
          },
          FlightSearchInterface: {
            template: `<div data-testid='flight-search-interface' />`,
          },
          AirportSearchInterface: {
            template: `<div data-testid='airport-search-interface' />`,
          },
        },
      })
    })

    test('should match the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    test('should show a SearchTypeSelector', () => {
      expect(wrapper.getByTestId('search-type-selector'))
    })

    test('should not show search interfaces', () => {
      expect(wrapper.queryByTestId('flight-search-interface')).toBe(null)
      expect(wrapper.queryByTestId('airport-search-interface')).toBe(null)
    })
  })
  describe('when search type is flights', () => {
    beforeEach(() => {
      state = {
        searchType: SEARCH_TYPE.FLIGHT,
      }
      store = new Store({
        state,
      })
      wrapper = renderWithVuetify(index, {
        store,
        mixins: [searchTypeMixin],
        stubs: {
          SearchTypeSelector: {
            template: `<div data-testid='search-type-selector' />`,
          },
          FlightSearchInterface: {
            template: `<div data-testid='flight-search-interface' />`,
          },
          AirportSearchInterface: {
            template: `<div data-testid='airport-search-interface' />`,
          },
        },
      })
    })

    test('should show a SearchTypeSelector', () => {
      expect(wrapper.getByTestId('search-type-selector'))
    })
    test('should show a FlightSearchInteface', () => {
      expect(wrapper.getByTestId('flight-search-interface'))
    })

    test('should not show AirportSearchInterface', () => {
      expect(wrapper.queryByTestId('airport-search-interface')).toBe(null)
    })
  })
  describe('when the search type is airport', () => {
    beforeEach(() => {
      state = {
        searchType: SEARCH_TYPE.AIRPORT,
      }
      store = new Store({
        state,
      })
      wrapper = renderWithVuetify(index, {
        store,
        mixins: [searchTypeMixin],
        stubs: {
          SearchTypeSelector: {
            template: `<div data-testid='search-type-selector' />`,
          },
          FlightSearchInterface: {
            template: `<div data-testid='flight-search-interface' />`,
          },
          AirportSearchInterface: {
            template: `<div data-testid='airport-search-interface' />`,
          },
        },
      })
    })

    test('should show a SearchTypeSelector', () => {
      expect(wrapper.getByTestId('search-type-selector'))
    })
    test('should show an AirportSearchInteface', () => {
      expect(wrapper.getByTestId('airport-search-interface'))
    })

    test('should not show FlightSearchInterface', () => {
      expect(wrapper.queryByTestId('flight-search-interface')).toBe(null)
    })
  })
})
