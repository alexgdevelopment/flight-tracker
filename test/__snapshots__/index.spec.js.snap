// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The "index" page when there is no search type selected should match the snapshot 1`] = `
"<div class=\\"row justify-center\\">
  <div class=\\"col col-9\\">
    <div data-testid=\\"search-type-selector\\"></div>
    <!---->
    <!---->
  </div>
</div>"
`;
